src/shared/types.ts

```diff
@@ -11,3 +11,11 @@ export interface Auth {
   signin: (token: string) => Promise<User>;
   signout: () => void;
 }
+
+export interface Theme {
+  toggleTheme?: (theme: Theme) => void;
```

Comment:
This part doesn't feel quite right. The hook probably shouldn't be returning the same interface that defines what a theme is.

Reply:
true